### PR TITLE
(gh-646) Create random keypair

### PR DESCRIPTION
**Pull request by mcanevet**

### CODE REVIEW

File: lib/beaker/hypervisor/openstack.rb [hunk 1]
```diff
@@ -91,6 +91,15 @@ def provision
         if @options[:openstack_keyname]
           @logger.debug "Adding optional key_name #{@options[:openstack_keyname]} to #{host.name} (#{host[:vmhostname]})"
           options[:key_name] = @options[:openstack_keyname]
+        else
+          @logger.debug "Generate a new rsa key"
```
Comment (anodelman) [inline lib/beaker/hypervisor/openstack.rb]: The openstack.provision method is getting pretty lengthy.  Let's get setting the keys as a method that can be called during provisioning to better encapsulate the code.

**Comment by mcanevet** [comment]
@anodelman done

File: lib/beaker/hypervisor/openstack.rb [hunk 2]
```diff
@@ -175,5 +176,25 @@ def enable_root(host)
       end
     end
 
+    # Get key_name from options or generate a new rsa key and add it to
```
Comment (anodelman) [inline lib/beaker/hypervisor/openstack.rb]: Missing @param for yard doc.

**Comment by anodelman** [comment]
ok to test

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by anodelman** [comment]
retest this please

**Comment by anodelman** [comment]
Failure unrelated to patch.

File: lib/beaker/hypervisor/openstack.rb (hunk 2 above)
Comment (anodelman) [inline lib/beaker/hypervisor/openstack.rb]: Code comment still valid.

**Comment by mcanevet** [comment]
@anodelman missing `@param` added

**Comment by raphink** [comment]
:+1:

File: lib/beaker/hypervisor/openstack.rb [hunk 3]
```diff
@@ -175,5 +176,26 @@ def enable_root(host)
       end
     end
 
+    #Get key_name from options or generate a new rsa key and add it to
+    #OpenStack keypairs
+    #
+    #@param [String] host The OpenStack host to provision
```
Comment (anodelman) [inline lib/beaker/hypervisor/openstack.rb]: Uh, a host is a Host object, not a string. :)

**Comment by mcanevet** [comment]
@anodelman fixed

**Comment by anodelman** [comment]
retest this please

**Comment by anodelman** [comment]
Known deb7 foss test failures.

**Pull request merged**